### ISSUE TITLE
[BUG] If a node attribute value contains an apostrophe, nx_translator().to_cypher_queries(G) breaks because the string values are contained in a single quote instead of double.

### ISSUE THREAD

**Issue by joshtracy3** [memgraph/gqlalchemy#318]
**Memgraph version** Which version did you use?
1.6.0

**Environment** Some information about the environment you are using Memgraph
on: operating system, how do you connect, with or without docker, which driver
etc.

Operating System: Darwin
OS Version: Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6000
OS Release: 23.6.0
OS Architecture: arm64
Full Platform Info: macOS-14.6.1-arm64-arm-64bit

Connecting with docker. Not sure about which driver

**Describe the bug** A clear and concise description of what the bug is.
When a string attribute value on a node contains an apostrophe, the nx_translator().to_cypher_queries(G) will break. It seems to be because string values are enclosed by a single quote instead of double quotes.

**To Reproduce** Steps to reproduce the behavior:

1. Run the following python code
```python
import networkx as nx
from gqlalchemy import Memgraph
from gqlalchemy.transformations.translators.nx_translator import NxTranslator

from warnings import filterwarnings

filterwarnings("ignore")

memgraph = Memgraph("127.0.0.1", 7687)
memgraph.drop_database()
graph = nx.Graph()
graph.add_nodes_from([(1, {"labels": "First", "name": "Josh's Laptop"}), 2, 3])
graph.add_edges_from([(1, 2, {"type": "EDGE_TYPE"}), (1, 3)])

nx_translator = NxTranslator()
for query in nx_translator.to_cypher_queries(graph):
    memgraph.execute(query)
```

**Expected behavior** A clear and concise description of what you expected to
happen.

I would expect that the queries should be seemlessly created and executed.

**Comment by katarinasupe** [comment]
Hi @joshtracy3 thank you for reporting this. It can definitely be handled better internally, but I tried running this and it worked:

`graph.add_nodes_from([(1, {"labels": "First", "name": "Josh\\'s Laptop"}), 2, 3])`

Let me know if it works for you. Btw. what are you working on?

**Comment by joshtracy3** [comment]
@katarinasupe yes this works if you escape the apostrophe's. There also seems to be an issue if the name contains a backslash. I just updated the functions `to_cypher_labels` and `to_cypher_values` in `nx_translator` to double quotes. Seems to work alright. I'm working on an entity linkage problem.

**Comment by katarinasupe** [comment]
Feel free to open a PR @joshtracy3 😄 Thank you for letting me know what works. I hope Memgraph is performing well for your use case. 

If you still haven't, [join our Discord](https://discord.gg/memgraph) server or schedule an [office hours call](https://memgraph.com/office-hours) for more questions.

**Comment by katarinasupe** [comment]
Hi @joshtracy3, if you'd like, you can contribute by solving the issue, and we can add it to the next release. A small token of appreciation from my side is Memgraph swag 😄

**Comment by james-geiger** [comment]
I've been investigating this issue as well, because I have a number of documents with newlines that are contextual and need to be kept in the representation of the text.

@joshtracy3's solution works for _certain_ fields, however I'm continuing to run into issues where there are a combination of characters in a string that need to be both escaped (e.g., an apostrophe `'`) and not escaped (e.g. newlines, `\n`).

I've traced the issue to this line [in the creation of cypher set commands ](https://github.com/memgraph/gqlalchemy/blob/76671ae6d5a829bba22bd6b6fe7aea60368f4b2a/gqlalchemy/models.py#L375)for a node.

@katarinasupe This feels like a related, but separate, issue, so will issue a PR solving the single apostrophe error but eager to hear thoughts on how to handle the "combination" issue.

**Comment by katarinasupe** [comment]
Hi @james-geiger, thank you for investigating this. I need to dive deeper into the issue and I'll get back to you. Thank for opening the PR as well 🙏 
If we manage to get this resolved, I would love to include it in the next release (March 12th). 

Btw. on what kind of project are you working on with Memgraph? We have a community on [Discord](https://discord.gg/memgraph) if you'd like to join + we can communicate faster there.
